linera-service/src/exporter/state.rs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc,
+};
+
+use linera_base::{crypto::CryptoHash, data_types::BlockHeight, identifiers::ChainId};
+use linera_client::config::DestinationId;
+use linera_sdk::{
+    ensure,
+    views::{RootView, View},
+};
+use linera_views::{
+    context::Context, log_view::LogView, map_view::MapView, register_view::RegisterView,
+    views::ClonableView,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::common::{BlockId, ExporterError, LiteBlockId};
+
+/// State of the linera exporter as a view.
+#[derive(Debug, RootView, ClonableView)]
+pub struct BlockExporterStateView<C> {
+    /// The causal state.
```

Comment:
OK, after reviewing the rest of the code I finally understand. I think the choice of _canonical_ is bit arbitrary but fine. Can you update the comment though? Maybe
```
/// Ordered collection of block hashes from all microchains, as indexed by the exporter.
```